render/gbm_allocator.c

```diff
@@ -159,7 +159,13 @@ static struct wlr_gbm_allocator *get_gbm_alloc_from_alloc(
 	return (struct wlr_gbm_allocator *)alloc;
 }
 
-struct wlr_allocator *wlr_gbm_allocator_create(int fd) {
+struct wlr_allocator *wlr_gbm_allocator_create(int drm_fd) {
+	int fd = fcntl(drm_fd, F_DUPFD_CLOEXEC, 0);
```

Comment:
We need to update the function doc-comment, which still states that the function takes ownership over the FD.